### PR TITLE
Allow nested objects

### PR DESCRIPTION
**Pull request by adrian-tiberius**

### CODE REVIEW

**Comment by cnizzardini** [comment]
Can you provide a bit more information as to what this accomplishes and what I might need to do to test it?

**Comment by adrian-tiberius** [comment]
This allows the use of nested objects inside swagger.yml file(have objects as schema properties):
https://swagger.io/docs/specification/data-models/data-types/#nested

File: src/Lib/Route/RouteScanner.php [hunk 1]
```diff
@@ -99,7 +99,7 @@ private function loadRoutes(): void
                 return strstr($fqn, '\\' . $prefix . $controller . 'Controller');
             });
 
-            if (count($results) == 1) {
+            if (count($results) >= 1) {
```
Comment (cnizzardini) [inline src/Lib/Route/RouteScanner.php:102]: Why was this change necessary?

**Comment by cnizzardini** [comment]
How are you using this? I am just not sure what the use-case is here and would like to gain a better understanding before merging this PR.

The library is already fairly agnostic about what you do in your YAML. For instance, testing the following over on the demo application without using your PR (just master) works fine: 

https://github.com/cnizzardini/cakephp-swagger-bake-demo

```yaml
    City:
      description: 'City with extended information including Country'
      type: object
      allOf:
        - $ref: '#/components/schemas/City'
        - type: object
          properties:
            country:
              $ref: '#/components/schemas/Country'
```

**Comment by adrian-tiberius** [comment]
I'm using this for a jsonapi api. I have responses structured like:

```yaml   
    Country:
      description: 'Country object'
      type: object
      properties:
        id:
          type: integer
        type:
          type: string
          example: 'countries'
        attributes:
          type: object
          properties:
            yitzo_country_code:
              type: string
              example: 'ES'
            is_active:
              type: boolean
              example: true
        relationships:
          type: object
          properties:
            description:
              type: array
              items:
                $ref: '#/components/schemas/CountryDescription'
```

File: src/Lib/Route/RouteScanner.php (hunk 1 above)
Comment (adrian-tiberius) [reply src/Lib/Route/RouteScanner.php:102]: For some reason, when using the cli commands and adding the namespaces['controllers'] setting, the config is parsed twice, resulting in 2 identical entries in $namespaces['controllers'] which leads to 2 identical entries in the $result array.

Comment (cnizzardini) [reply src/Lib/Route/RouteScanner.php:102]: Can you paste in your `swagger_bake.php` config?

**Comment by cnizzardini** [comment]
> I'm using this for a jsonapi api. I have responses structured like:
> 
> ```yaml
>     Country:
>       description: 'Country object'
>       type: object
>       properties:
>         id:
>           type: integer
>         type:
>           type: string
>           example: 'countries'
>         attributes:
>           type: object
>           properties:
>             yitzo_country_code:
>               type: string
>               example: 'ES'
>             is_active:
>               type: boolean
>               example: true
>         relationships:
>           type: object
>           properties:
>             description:
>               type: array
>               items:
>                 $ref: '#/components/schemas/CountryDescription'
> ```

Thanks, I was able to replicate this.

File: src/Lib/Route/RouteScanner.php (hunk 1 above)
Comment (adrian-tiberius) [reply src/Lib/Route/RouteScanner.php:102]:
```php
return [
    'SwaggerBake' => [
        'prefix' => '/api-v3',
        'yml' => '/config/swagger.yml',
        'json' => '/webroot/swagger.json',
        'webPath' => '/swagger.json',
        'hotReload' => true, //\Cake\Core\Configure::read('debug'),

        'namespaces' => [
            'controllers' => ['\ApiV3\\'],
            'entities' => ['\SharedModels3\\'],
            'tables' => ['\SharedModels3\\']
        ]
    ]
];
```

Comment (cnizzardini) [reply src/Lib/Route/RouteScanner.php:102]: Reference this pull request I just opened: https://github.com/cnizzardini/cakephp-swagger-bake/pull/276

I think it likely solves these issues, but I wasn't exactly sure how to push into your PR or fork. That's beyond my gitfu.

Comment (adrian-tiberius) [reply src/Lib/Route/RouteScanner.php:102]: Yes, I tested that pull request and it works as expected.

Comment (cnizzardini) [reply src/Lib/Route/RouteScanner.php:102]: Okay, we'll close this PR out and use that one. I should be able to get a release out in the next day or two after I regression test a bit more.
